### PR TITLE
ECP-8673/create pending invoice on authorisations for manual capture

### PR DESCRIPTION
**Pull request by hossam-adyen**
**Description**

For manual capture orders we currently leave orders in pending_payment status until the order is captured.
This results in a situation where an order can be cancelled by magento before a capture webhook is received and processed.
According to [Magento's order status flow](https://business.adobe.com/blog/basics/magento-order-status) a payment in this state should be processing

**Tested scenarios**

Handle successful authorisation with auto capture and manual capture => Test/Unit/Helper/Webhook/AuthorisationWebhookHandlerTest.php

Fixes  
https://github.com/Adyen/adyen-magento2/issues/2097

### CODE REVIEW

File: Helper/Webhook/AuthorisationWebhookHandler.php [hunk 1]
```diff
@@ -117,12 +117,14 @@ private function handleSuccessfulAuthorisation(Order $order, Notification $notif
         $isFullAmountAuthorized = $this->adyenOrderPaymentHelper->isFullAmountAuthorized($order);
 
         if ($isFullAmountAuthorized) {
-            $order = $this->orderHelper->setPrePaymentAuthorized($order);
+            $order = $this->orderHelper->setPrePaymentAuthorized($order); //here we set the order as pending payment in case of customer set manual capture status to it!
```
Comment (candemiralp) [inline Helper/Webhook/AuthorisationWebhookHandler.php:120]: Can you please remove or update the comment? It seems irrelevant after the fix.

**Comment by peterojo** [comment]
@hossam-adyen could you please add a description using the usual template?

**Review by candemiralp — APPROVED** [review]

File: Setup/Patch/Data/PreAuthorizedSettingsMigration.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    /**
+     * Ensure that new path does not exist before updating
+     *
+     * @param ModuleDataSetupInterface $setup
+     */
+    public function updateSchemaVersion(ModuleDataSetupInterface $setup)
+    {
+        $path = 'payment/adyen_abstract/';
```
Comment (hossam-adyen) [inline Setup/Patch/Data/PreAuthorizedSettingsMigration.php:55]:
```suggestion
        $path = 'payment/adyen_abstract/payment_pre_authorized';
```

**Review by RokPopov — APPROVED** [review]

**Review by khushboo-singhvi — APPROVED** [review]
